File: android/src/main/java/com/indigo/plugins/uptime/UptimePlugin.java
package com.indigo.plugins.uptime;

import android.os.SystemClock;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.PluginMethod;
@CapacitorPlugin(name = "Uptime")
public class UptimePlugin extends Plugin {

    @PluginMethod
    public void getUptime(PluginCall call) {
        boolean includeDeepSleep = call.getBoolean("includeDeepSleep", false);

        long uptime = includeDeepSleep ? SystemClock.elapsedRealtime() : SystemClock.uptimeMillis();

        JSObject result = new JSObject();
        result.put("uptime", uptime);

        call.resolve(result);
    }
}

File: ios/Sources/UptimePlugin/UptimePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UptimePlugin)
public class UptimePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "UptimePlugin"
    public let jsName = "Uptime"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Uptime()

   @objc func getUptime(_ call: CAPPluginCall) {
        let includeDeepSleep = call.getBool("includeDeepSleep") ?? false
        let uptime = ProcessInfo.processInfo.systemUptime

        let result: Double = includeDeepSleep ? uptime : uptime // Adjust for different logic if needed
        call.resolve(["uptime": result * 1000]) // Convert to milliseconds
    }
}
